src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -218,7 +218,7 @@ public static void CompileMethodBodies(
 
         // Returns the MethodSymbol for the assembly entrypoint.  If the user has a Task returning main,
         // this function returns the synthesized Main MethodSymbol.
-        private static MethodSymbol GetEntryPoint(CSharpCompilation compilation, PEModuleBuilder moduleBeingBuilt, bool hasDeclarationErrors, bool emitMethodBodies, BindingDiagnosticBag diagnostics, CancellationToken cancellationToken)
+        internal static MethodSymbol GetEntryPoint(CSharpCompilation compilation, PEModuleBuilder moduleBeingBuilt, bool hasDeclarationErrors, bool emitMethodBodies, BindingDiagnosticBag diagnostics, CancellationToken cancellationToken)
```

Comment:
>GetEntryPoint [](http://example.com/codeflow?start=37&length=13)

It looks like this function still performs a lot of work around body (creating, lowering, etc.) even when ```emitMethodBodies``` is `false`. Therefore, its effect in our scenario feels insufficient. It feels like additional changes to the body of this method are necessary to handle scenario when ```moduleBeingBuilt.EmitOptions.EmitMetadataOnly``` is `true`.